### PR TITLE
Add helper trait to provide type-hinting for zend-view

### PR DESCRIPTION
**Pull request by thexpand**
In favor of the following proposal:
zendframework/zend-view#102

### CODE REVIEW

**Comment by thexpand** [comment]
@froschdesign 
I'm not sure about this one, because there is already a doc-block `@method` in `Zend\View\Renderer\PhpRenderer`. Take a look:
https://github.com/zendframework/zend-view/blob/master/src/Renderer/PhpRenderer.php#L45

File: src/View/HelperTrait.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?php
+
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: Doc block with copyright and licence is missing. An example can be found here:

https://github.com/zendframework/maintainers/blob/abab4a99f3e6d1fac4d7400232ce30ee0bc95249/template/src/ConfigProvider.php#L2-L6

File: src/View/HelperTrait.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Trait HelperTrait
+ *
+ * The trait provides convenience methods for view helpers,
+ * defined by the zend-mvc-plugin-flashmessenger component.
+ * It is designed to be used for type-hinting $this variable
+ * inside zend-view templates via doc blocks.
+ *
+ * The base class is PhpRenderer, followed by the helper trait from
+ * the zend-mvc-plugin-flashmessenger component. However, multiple helper traits
+ * from different Zend components can be chained afterwards.
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: "Zend components" is wrong, these are "Zend Framework components".

File: src/View/HelperTrait.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Zend\Mvc\Plugin\FlashMessenger\View;
+
+use Zend\Mvc\Plugin\FlashMessenger\FlashMessenger as PluginFlashMessenger;
+use Zend\Mvc\Plugin\FlashMessenger\View\Helper\FlashMessenger;
+
+/**
+ * Trait HelperTrait
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: "Trait HelperTrait" – what is this?! 😉
Can we find a short and meaningful name / description?

File: src/View/HelperTrait.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * It is designed to be used for type-hinting $this variable
+ * inside zend-view templates via doc blocks.
+ *
+ * The base class is PhpRenderer, followed by the helper trait from
+ * the zend-mvc-plugin-flashmessenger component. However, multiple helper traits
+ * from different Zend components can be chained afterwards.
+ *
+ * @example @var \Zend\View\Renderer\PhpRenderer|\Zend\Mvc\Plugin\FlashMessenger\View\HelperTrait $this
+ *
+ * @method FlashMessenger|PluginFlashMessenger flashMessenger($namespace = null)
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: We can add also the data types. See my comment here: https://github.com/zendframework/zend-i18n/pull/102#discussion_r215139655

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Comment by froschdesign** [comment]
The usage of this trait must be explained in the documentation.

**Comment by froschdesign** [comment]
Btw. Please use the `develop` branch for new features.

https://github.com/zendframework/zend-mvc-plugin-flashmessenger/blob/master/docs/CONTRIBUTING.md#what-branch-to-issue-the-pull-request-against

**Comment by thexpand** [comment]
@froschdesign 
I suggest adding the docs as a sub-section of the main section `View Helper`.
Maybe at the bottom?
See more about the naming in my comment here: https://github.com/zendframework/zend-i18n/pull/102#issuecomment-418711168
Here's a link to the main section in the docs: https://docs.zendframework.com/zend-mvc-plugin-flashmessenger/flash-messenger-view-helper/

**Comment by froschdesign** [comment]
@thexpand 
> Maybe at the bottom?

For this single page is bottom okay.

File: src/View/HelperTrait.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-mvc-plugin-flashmessenger for the canonical source repository
+ * @copyright Copyright (c) 2005-2018 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (froschdesign) [inline src/View/HelperTrait.php]: Please use for new files only the current year.

**Comment by thexpand** [comment]
@froschdesign 
I've placed the docs at the bottom, without block quotes, in contrast to the other PRs (https://github.com/zendframework/zend-i18n/pull/102, https://github.com/zendframework/zend-form/pull/217)

**Review by michalbundyra — APPROVED** [review]

**Comment by michalbundyra** [comment]
Thanks, @thexpand!

**Pull request merged**